users/models.py

```diff
@@ -1,3 +1,385 @@
+        if role is not None and role not in dict(cls.Role.choices):
+            raise ValueError("Invalid role value")
+        if not password or not isinstance(password, str) or len(password) < 8:
+            raise ValueError("Password must be a string at least 8 characters long")
+
+        user = cls(
+            email=email,
+            first_name=first_name,
+            last_name=last_name,
+            user_phone=user_phone,
+            title=title,
+            role=role or cls.Role.USER,
+            **other_fields
+        )
+        user.set_password(password)
+        try:
+            user.save()
+        except (IntegrityError, DataError) as e:
+            raise ValueError(f"Database error: {e}")
+        return user
+
+    def update(self, **kwargs):
+        """
+        Update user fields with validation.
+
+        Args:
+            **kwargs: Fields to update.
+
+        Raises:
+            ValueError: If validation fails.
+
+        Returns:
+            User: Updated user instance.
+        """
+
+        def validate_first_name(v):
+            return isinstance(v, str) and len(v) <= 50
+
+        def validate_last_name(v):
+            return isinstance(v, str) and len(v) <= 50
+
+        def validate_user_phone(v):
+            return v is None or (isinstance(v, str) and len(v) <= 20)
+
+        def validate_title(v):
+            return v is None or (isinstance(v, str) and len(v) <= 100)
+
+        def validate_role(v):
+            return v in dict(self.Role.choices)
+
+        def validate_is_active(v):
+            return isinstance(v, bool)
+
+        def validate_is_staff(v):
+            return isinstance(v, bool)
+
+        def validate_email_field(v):
+            if not isinstance(v, str) or len(v) > 50:
+                return False
+            try:
+                validate_email(v)
+                return True
+            except ValidationError:
+                return False
+
+        allowed_fields = {
+            'first_name': validate_first_name,
+            'last_name': validate_last_name,
+            'user_phone': validate_user_phone,
+            'title': validate_title,
+            'role': validate_role,
+            'email': validate_email_field,
+            'is_active': validate_is_active,
+            'is_staff': validate_is_staff
+        }
+
+        for attr, validator in allowed_fields.items():
+            if attr in kwargs:
+                value = kwargs[attr]
+                if not validator(value):
+                    raise ValueError(f"Invalid value for field '{attr}': {value}")
+                setattr(self, attr, value)
+
+        if 'password' in kwargs:
+            password = kwargs['password']
+            if not isinstance(password, str) or len(password) < 8 or len(password) > 128:
+                raise ValueError("Password must be a string between 8 and 128 characters.")
+            self.set_password(password)
+
+        self.save()
+        return self
```

Comment:
When updating the password, you are setting it without checking if it's unchanged.
If password is not in kwargs or is None, maybe skip set_password to avoid unnecessary hash calls.